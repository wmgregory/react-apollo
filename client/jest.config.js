module.exports = {
  roots: ['<rootDir>/src/'],

  collectCoverage: true,
  collectCoverageFrom: ['src/**/*.{tsx?}', '!src/**/*.d.ts', '!src/**/*.mock.ts', '!src/**/*.js'],
  coverageReporters: ['clover', 'json', 'lcov', 'text-summary'],
  coverageThreshold: {
    global: {
      statements: 75,
      branches: 75,
      functions: 75,
      lines: 75,
    },
  },
  coveragePathIgnorePatterns: ['/config/', '/node_modules/'],

  setupFilesAfterEnv: ['<rootDir>/config/jest/setupJest.ts'],
  snapshotSerializers: ['enzyme-to-json/serializer'],
};
