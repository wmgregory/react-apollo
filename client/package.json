{
  "name": "client",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "snyk-protect": "yarn snyk protect",
    "prepublish": "yarn snyk-protect",
    "test": "react-scripts test --watchAll=false --coverage",
    "eject": "react-scripts eject"
  },
  "dependencies": {
    "apollo-boost": "0.4.3",
    "graphql": "14.4.2",
    "react": "^16.8.6",
    "react-apollo": "2.5.8",
    "react-dom": "^16.8.6",
    "react-scripts": "4.0.0",
    "snyk": "^1.429.0"
  },
  "devDependencies": {
    "@types/enzyme": "3.10.2",
    "@types/enzyme-adapter-react-16": "1.0.5",
    "@types/jest": "24.0.15",
    "@types/node": "12.6.2",
    "@types/react": "16.8.23",
    "@types/react-dom": "16.8.4",
    "enzyme": "3.10.0",
    "enzyme-adapter-react-16": "1.14.0",
    "enzyme-to-json": "3.3.5",
    "jest-enzyme": "7.0.2",
    "typescript": "3.5.3"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "snyk": true
}