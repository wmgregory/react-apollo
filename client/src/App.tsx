import React from 'react';

import './App.css';

import { Header } from './components';

const App: React.FC = () => {
  return (
    <div className='App'>
      <Header title='React Apollo demo' />
    </div>
  );
};

export default App;
