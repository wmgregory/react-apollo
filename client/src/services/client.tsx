import ApolloClient from 'apollo-boost';

const GQL_CLIENT = {
  uri: 'http://localhost:4000/graphql',
};

export const client = new ApolloClient(GQL_CLIENT);
