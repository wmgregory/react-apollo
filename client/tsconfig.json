{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "noImplicitAny": true,
    "noImplicitThis": true,
    "strictNullChecks": true,
    "jsx": "preserve"
  },
  "paths": {
    "*": ["types/*"],
    "#components": ["src/components"],
    "#components/*": ["src/components/*"],
    "#constants": ["src/constants"],
    "#constants/*": ["src/constants/*"],
    "#pages": ["src/pages"],
    "#pages/*": ["src/pages/*"],
    "#types": ["src/types"],
    "#types/*": ["src/types/*"]
  },
  "include": ["src"]
}
