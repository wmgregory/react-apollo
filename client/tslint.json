{
  "extends": ["tslint-config-prettier"],
  "rules": {
    "no-implicit-dependencies": false, //https://github.com/palantir/tslint/issues/3364
    "no-submodule-imports": false,
    "object-literal-shorthand": true,
    "quotemark": [true, "single", "jsx-single"],
    "typedef": [true, "arrow-call-signature", "member-variable-declaration", "property-declaration", "parameter"],
    "whitespace": [true, "check-branch", "check-decl", "check-operator", "check-module", "check-rest-spread"],
    "newline-before-return": true
  }
}
