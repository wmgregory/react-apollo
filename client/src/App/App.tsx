import React from 'react';
import { ApolloProvider } from '@apollo/react-hooks';

import './App.css';

import { Header } from '../components';
import { client } from '../services/client';

export const App: React.FC = () => {
  return (
    <ApolloProvider client={client}>
      <div className="App">
        <Header title="React Apollo demo" />
      </div>
    </ApolloProvider>
  );
};
