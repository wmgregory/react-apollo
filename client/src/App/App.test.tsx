import React from 'react';

import { shallow } from 'enzyme';

import * as client from '../services/client';
import { App } from './App';

describe('App', () => {
  beforeEach(() => {
    jest.spyOn(client, 'client').mockReturnValue({});
  });
  it('renders without crashing', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toBeDefined();
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
