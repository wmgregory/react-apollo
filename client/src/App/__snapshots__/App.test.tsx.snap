// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
"<ApolloProvider client={{...}}>
  <div className=\\"App\\">
    <Header title=\\"React Apollo demo\\" />
    <RandomNumber />
  </div>
</ApolloProvider>"
`;
